src/ArrowFunctions.cpp

```diff
@@ -188,18 +191,20 @@ int cpp_readColumnByName(const char* filename, void* chpl_arr, const char* colna
       auto chpl_ptr = (int64_t*)chpl_arr;
       parquet::Int64Reader* reader =
         static_cast<parquet::Int64Reader*>(column_reader.get());
+      reader->Skip(startIdx);
 
-      while (reader->HasNext()) {
+      while (reader->HasNext() && i < numElems) {
         (void)reader->ReadBatch(batchSize, nullptr, nullptr, &chpl_ptr[i], &values_read);
```

Comment:
If you don't I think you can overrun the end of a local array. Like if you're reading in a 20 element file across 2 nodes, I think node 1 would still readying in batchSize (stopping at the end of the file, so reading in 20 elements total) instead of just 10 elements

Reply:
Ya, that is what I was thinking as well, but took me a minute to get there.